### PR TITLE
Minor fixes in the VQA tutorial

### PR DESCRIPTION
**Pull request by NarineK**
+ Added Early Forward Termination Exception for the resnet model after accessing the 4th layer and tested that it works properly with DataParallel

+ Tested and added proper support for DataParallel that is being executed on all GPUs. Before we used to fix it to 2 devices only.

### CODE REVIEW

File: tutorials/Multimodal_VQA_Interpret.ipynb [hunk 1]
```diff
@@ -230,13 +252,13 @@
...
   {
    "cell_type": "code",
-   "execution_count": 11,
+   "execution_count": 9,
    "metadata": {},
    "outputs": [],
    "source": [
     "vqa_resnet = VQA_Resnet_Model(vqa_net.module.text.embedding.num_embeddings)\n",
     "# `device_ids` contains a list of GPU ids which are used for paralelization supported by `DataParallel`\n",
-    "vqa_resnet = torch.nn.DataParallel(vqa_resnet, device_ids=[0,1])\n",
```
Comment (NarineK) [inline tutorials/Multimodal_VQA_Interpret.ipynb]: Removing fixed number of devices in order to be able to take advantage of all of them.

File: tutorials/Multimodal_VQA_Interpret.ipynb [hunk 2]
```diff
@@ -134,7 +128,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "vqa_net = torch.nn.DataParallel(Net(num_tokens), device_ids=[0,1])\n",
+    "vqa_net = torch.nn.DataParallel(Net(num_tokens))\n",
```
Comment (NarineK) [inline tutorials/Multimodal_VQA_Interpret.ipynb]: Removing fixed number of devices in order to be able to take advantage of all of them.

File: tutorials/Multimodal_VQA_Interpret.ipynb [hunk 3]
```diff
@@ -191,16 +204,25 @@
...
     "        self.r_model.to(device)\n",
-    "        self.buffer = None\n",
+    "\n",
+    "        self.buffer = {}\n",
+    "        lock = threading.Lock()\n",
     "\n",
     "        def save_output(module, input, output):\n",
-    "            self.buffer = output\n",
+    "            with lock:\n",
+    "                self.buffer[output.device]= output\n",
```
Comment (miguelmartin75) [inline tutorials/Multimodal_VQA_Interpret.ipynb]: nit: space before `=`

**Comment by miguelmartin75** [comment]
LGTM (beside the nit)
